text-generation-inference/tgi_env.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import os
+import sys
+from typing import Any, Dict, List, Optional
+
+from huggingface_hub import constants
+from transformers import AutoConfig
+
+from optimum.neuron.modeling_decoder import get_available_cores
+from optimum.neuron.utils import get_hub_cached_entries
+from optimum.neuron.utils.version_utils import get_neuronxcc_version
+
+
+logger = logging.getLogger(__name__)
+
+tgi_router_env_vars = ["MAX_BATCH_SIZE", "MAX_TOTAL_TOKENS", "MAX_INPUT_LENGTH"]
+tgi_server_env_vars = ["HF_BATCH_SIZE", "HF_SEQUENCE_LENGTH", "HF_NUM_CORES", "HF_AUTO_CAST_TYPE"]
+
+env_config_peering = [
+    ("MAX_BATCH_SIZE", "batch_size"),
+    ("MAX_TOTAL_TOKENS", "sequence_length"),
+    ("HF_BATCH_SIZE", "batch_size"),
+    ("HF_AUTO_CAST_TYPE", "auto_cast_type"),
+    ("HF_SEQUENCE_LENGTH", "sequence_length"),
+    ("HF_NUM_CORES", "num_cores"),
+]
+
+# By the end of this script all env var should be specified properly
+env_vars = tgi_server_env_vars + tgi_router_env_vars
+
+available_cores = get_available_cores()
+neuronxcc_version = get_neuronxcc_version()
+
+
+def parse_cmdline(argv: List[str] = None) -> argparse.Namespace:
```

Comment:
This function actually aligns the env variables also, so I think its name should reflect that too.